src/Npgsql/NpgsqlCommand.cs

```diff
@@ -665,13 +663,13 @@ public void Unprepare()
             using (connector.StartUserAction())
             {
                 var sendTask = SendClose(connector, false);
-                foreach (var statement in _statements.Where(s => s.PreparedStatement?.State == PreparedState.BeingUnprepared))
-                {
-                    Expect<CloseCompletedMessage>(connector.ReadMessage(), connector);
-                    Debug.Assert(statement.PreparedStatement != null);
-                    statement.PreparedStatement.CompleteUnprepare();
-                    statement.PreparedStatement = null;
-                }
+                foreach (var statement in _statements)
+                    if (statement.PreparedStatement?.State == PreparedState.BeingUnprepared)
```

Comment:
Same as above - I don't think we should be micro-optimizing non-sensitive methods like this one...

Reply:
It's not a micro. In that case `PreparedStatement` is checked for `null` and the compiler can detect this which is impossible with LINQ.

Reply:
So then why not use OfType like you did elsewhere?